packages/deployments/contracts/contracts/lib/Connext/ConnextLogic.sol

```diff
@@ -864,6 +878,177 @@ library ConnextLogic {
+
+    // if the token is of remote origin, mint the tokens. will either
+    // - be credited to router (fast liquidity)
+    // - be reserved for execution (slow liquidity)
+    if (!_args.tokenRegistry.isLocalOrigin(token)) {
+      IBridgeToken(token).mint(address(this), amount);
+      // Tell the token what its detailsHash is
+      IBridgeToken(token).setDetailsHash(details);
+    }
+    // NOTE: if the token is of local origin, it means it was escrowed
+    // in this contract at xcall
+
+    // mark the transfer as reconciled
+    _reconciledTransfers[transferId] = true;
+
+    return (amount, token, transferId);
+  }
+
+  /**
+   * @notice Repays to Aave Portal if the transfer was executed with fast path using Portal liquidity
+   * @return The amount left after the repayment
+   * @return The address of the adopted asset
+   */
+  function _reconcileProcessPortal(
+    ReconcileArgs memory _args,
+    uint256 _amount,
+    address _token,
+    bytes32 _transferId,
+    mapping(bytes32 => uint256) storage _aavePortalsTransfers,
+    mapping(bytes32 => IStableSwap) storage _adoptedToLocalPools,
+    mapping(bytes32 => address) storage _canonicalToAdopted
+  ) private returns (uint256, address) {
+    ReconcilePortalVars memory vars;
+
+    vars.portalTransferAmount = _aavePortalsTransfers[_transferId];
+
+    if (vars.portalTransferAmount == 0) return (_amount, _token);
+
+    // Swap all the transfer amount from local asset to adopted if needed
+    (vars.adoptedAmountOut, vars.adopted) = AssetLogic.swapFromLocalAssetIfNeeded(
+      _canonicalToAdopted,
+      _adoptedToLocalPools,
+      _args.tokenRegistry,
+      _token,
+      _amount
+    );
+
+    // Calculates the amount to be repaid to the portal
+    (vars.totalRepayAmount, vars.backUnbackedAmount, vars.portalFee) = _calculatePortalRepayment(
+      vars.adoptedAmountOut,
+      vars.portalTransferAmount,
+      _args.portalFeeNumerator,
+      _args.portalFeeDenominator
+    );
+
+    // Edge case with some tokens: Example USDT in ETH Mainnet, after the backUnbacked call there could be a remaining allowance if not the whole amount is pulled by aave.
+    // Later, if we try to increase the allowance it will fail. USDT demands if allowance is not 0, it has to be set to 0 first.
+    // TODO: Should we call approve(0) and approve(totalRepayAmount) instead? or with a try catch to not affect gas on all cases?
+    // Example: https://github.com/aave/aave-v3-periphery/blob/ca184e5278bcbc10d28c3dbbc604041d7cfac50b/contracts/adapters/paraswap/ParaSwapRepayAdapter.sol#L138-L140
+    SafeERC20Upgradeable.safeIncreaseAllowance(IERC20Upgradeable(vars.adopted), _args.aavePool, vars.totalRepayAmount);
+
+    // TODO: What if backUnbacked call fails? For example if the asset is inactive, paused or frozen?
+    // What is the impact of a reconcile to be reverted? Can be re-executed later?
+    IAavePool(_args.aavePool).backUnbacked(vars.adopted, vars.backUnbackedAmount, vars.portalFee);
```

Comment:
hmm if a message is handled (i.e. reconciled is called), it doesn't get rehandled by nomad and can't really be re-processed easily (i will have to double check this to be sure). we should probably have a separate method to repay the loans in case this fails (or if there is a problem with nomad so you can prevent getting into the hole by not closing out a position)

Reply:
Oh, I see. Then, I think it is important to correctly handle any external call failure to not revert the reconcile call and have a separate method to repay in case of any of these mentioned issues with Aave or Nomad.